site/src/pages/TaskPage/TaskAppIframe.tsx

```diff
@@ -7,55 +6,43 @@ import {
 	DropdownMenuTrigger,
 } from "components/DropdownMenu/DropdownMenu";
 import { Spinner } from "components/Spinner/Spinner";
+import { useProxy } from "contexts/ProxyContext";
 import { EllipsisVertical, ExternalLinkIcon, HouseIcon } from "lucide-react";
-import { useAppLink } from "modules/apps/useAppLink";
-import type { Task } from "modules/tasks/tasks";
+import { type AppLink, useAppLink } from "modules/apps/useAppLink";
+import type { Task, WorkspaceAppWithAgent } from "modules/tasks/tasks";
 import { type FC, useRef } from "react";
 import { Link as RouterLink } from "react-router";
 import { cn } from "utils/cn";
+import { TaskWildcardWarning } from "./TaskWildcardWarning";
 
 type TaskAppIFrameProps = {
 	task: Task;
-	app: WorkspaceApp;
+	app: WorkspaceAppWithAgent;
 	active: boolean;
-	pathname?: string;
 };
 
 export const TaskAppIFrame: FC<TaskAppIFrameProps> = ({
 	task,
 	app,
 	active,
-	pathname,
 }) => {
-	const agent = task.workspace.latest_build.resources
-		.flatMap((r) => r.agents)
-		.filter((a) => !!a)
-		.find((a) => a.apps.some((a) => a.id === app.id));
-
-	if (!agent) {
-		throw new Error(`Agent for app ${app.id} not found in task workspace`);
-	}
-
 	const link = useAppLink(app, {
-		agent,
+		agent: app.agent,
 		workspace: task.workspace,
 	});
-
-	const appHref = (): string => {
```

Comment:
ooooh I see, it used to be here. still, I'm really not convinced that this actually does anything, and if it does do anything, it needs a comment documenting what that is in the new location.